rasa/nlu/components.py

```diff
@@ -91,13 +91,35 @@ def validate_requirements(component_names: List[Optional[Text]]) -> None:
         )
 
 
+def validate_component_keys(
+    component: "Component", component_config: Dict[Text, Any]
+) -> None:
+    """Validates that all keys for a component are valid.
+
+    Args:
+        component: The component class
+        component_config: The user-provided config for the component in the pipeline
+    """
+    component_name = component_config.get("name")
+    allowed_keys = set(component.defaults.keys())
+    provided_keys = set(component_config.keys())
+    provided_keys = (key for key in provided_keys if key != "name")
```

Comment:
Good catch! That's also working
```suggestion
    provided_keys.discard("name")
```